### ISSUE TITLE
javacript code formatting not working after typescript v2.x update (from vscode)

### ISSUE THREAD

**Issue by mlakmal** [Microsoft/TypeScript#11199]
TypeScript Version: 2.0.x
VS Code insider 1.6.x 
Steps to Reproduce:

updated to latest VS Code insider v1.6.0 and with typescript v2.0.x
format the javascript code with below workspace settings. 
"editor.tabSize": 2, "javascript.format.insertSpaceAfterKeywordsInControlFlowStatements": true, "javascript.format.insertSpaceAfterFunctionKeywordForAnonymousFunctions": true, "javascript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis": false, "javascript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets": false,
    
above formatting settings are not adhered after typescript v2.0x update

tabsize is 4 and not using above override and spaces are not getting added after function keyword and parenthesis/brackets. for tabsize and  spaces

Actual behavior: should support overrides

https://github.com/Microsoft/vscode/issues/12705

**Comment by dbaeumer** [comment]
This is a real issue has the screen cast shows:
![cast](https://cloud.githubusercontent.com/assets/1931590/18907886/bd80f6ea-856d-11e6-9bb0-3ce93fae47c3.gif)

Enabling tracing shows that although configured differently the formatter replaces tabs with spaces.

```
[Trace - 11:01:20 AM] Sending request: projectInfo (3). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "p:\\mseng\\VSCode\\Playgrounds\\bugs\\12705\\test.js",
    "needFileNameList": true
}
[Trace - 11:01:20 AM] Response received: projectInfo (3). Request took 1 ms. Success: true 
Result: {
    "languageServiceDisabled": false,
    "fileNames": [
        "c:/Program Files (x86)/Microsoft VS Code Insiders/resources/app/extensions/typescript/server/typescript/lib/lib.d.ts",
        "p:/mseng/VSCode/Playgrounds/bugs/12705/test.js"
    ]
}
[Trace - 11:01:24 AM] Sending request: configure (4). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "p:\\mseng\\VSCode\\Playgrounds\\bugs\\12705\\test.js",
    "formatOptions": {
        "tabSize": 2,
        "indentSize": 2,
        "convertTabsToSpaces": false,
        "newLineCharacter": "\n",
        "insertSpaceAfterCommaDelimiter": true,
        "insertSpaceAfterSemicolonInForStatements": true,
        "insertSpaceBeforeAndAfterBinaryOperators": true,
        "insertSpaceAfterKeywordsInControlFlowStatements": true,
        "insertSpaceAfterFunctionKeywordForAnonymousFunctions": true,
        "insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis": false,
        "insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets": false,
        "placeOpenBraceOnNewLineForFunctions": false,
        "placeOpenBraceOnNewLineForControlBlocks": false
    }
}
[Trace - 11:01:24 AM] Response received: configure (4). Request took 1 ms. Success: true 
[Trace - 11:01:24 AM] Sending request: format (5). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "p:\\mseng\\VSCode\\Playgrounds\\bugs\\12705\\test.js",
    "line": 1,
    "offset": 1,
    "endLine": 6,
    "endOffset": 2
}
[Trace - 11:01:24 AM] Response received: format (5). Request took 42 ms. Success: true 
Result: [
    {
        "start": {
            "line": 2,
            "offset": 1
        },
        "end": {
            "line": 2,
            "offset": 2
        },
        "newText": "    "
    },
    {
        "start": {
            "line": 2,
            "offset": 4
        },
        "end": {
            "line": 2,
            "offset": 4
        },
        "newText": " "
    },
    {
        "start": {
            "line": 3,
            "offset": 1
        },
        "end": {
            "line": 3,
            "offset": 2
        },
        "newText": "    "
    },
    {
        "start": {
            "line": 4,
            "offset": 1
        },
        "end": {
            "line": 4,
            "offset": 2
        },
        "newText": "    "
    },
    {
        "start": {
            "line": 4,
            "offset": 7
        },
        "end": {
            "line": 4,
            "offset": 7
        },
        "newText": " "
    },
    {
        "start": {
            "line": 4,
            "offset": 8
        },
        "end": {
            "line": 4,
            "offset": 8
        },
        "newText": " "
    },
    {
        "start": {
            "line": 5,
            "offset": 1
        },
        "end": {
            "line": 5,
            "offset": 2
        },
        "newText": "    "
    }
]
```

**Comment by dbaeumer** [comment]
@mhegazy @zhengbli could you please look into this and consider a 2.0.4 with a fix. This is a bad regression in terms for formatting TS/JS files.

**Comment by dbaeumer** [comment]
@mhegazy @zhengbli: this was also a problem with us moving to the next version of Electron to overcome the real path issue which node addressed in 6.4.0. After removing the work around code in VS Code for this things look ok now.
